soc/riscv/esp32c3/Kconfig.soc

```diff
@@ -11,6 +11,9 @@ config SOC_ESP32C3
 	select PINCTRL
 	select XIP
 	select HAS_ESPRESSIF_HAL
+	select RISCV_ISA_RV32I
+	select RISCV_ISA_EXT_M
+	select RISCV_ISA_EXT_A
```

Comment:
esp32c3 does not have the atomic instructions.

Reply:
this was already discussed in https://github.com/zephyrproject-rtos/zephyr/pull/45582#discussion_r875268255 . This was already broken before this PR.

Reply:
Got it, thanks for the clarification @carlocaione